src/components/utils/onfidoApi.js

```diff
@@ -3,44 +3,34 @@ import Tracker from '../../Tracker'
 import forEach from 'object-loops/for-each'
 import { humanizeField } from '../utils'
 
-const formatError = ({response, status}) => ({status, response: JSON.parse(response)})
+const formatError = ({response, status}, onError) =>
+  onError({status, response: JSON.parse(response)})
 
-export const postToOnfido = ({blob, documentType, side}, captureType, token, advanced_validation, onSuccess, onError) => {
-  if (captureType === 'face') {
-    return sendFile(
-      {blob},
-      'live_photos',
-      token,
-      onSuccess,
-      request => onError(formatError(request))
-    )
-  }
-  sendFile(
-    {blob, type: documentType, side, advanced_validation},
-    'documents',
-    token,
-    onSuccess,
-    request => onError(formatError(request))
-  )
+export const uploadDocument = (data, token, onSuccess, onError) => {
+  const endpoint = `${process.env.ONFIDO_API_URL}/v2/documents`
+  sendFile(endpoint, data, token, onSuccess, onError)
+}
+
+export const uploadLivePhoto = (data, token, onSuccess, onError) => {
+  const endpoint = `${process.env.ONFIDO_API_URL}/v2/live_photos`
+  sendFile(endpoint, data, token, onSuccess, onError)
 }
 
+
 const objectToFormData = (object) => {
   const formData = new FormData()
   forEach(object, (value, key) => formData.append(key, value))
   return formData
 }
 
-const sendFile = ({blob, ...extraOptions}, path, token, onSuccess, onError) => {
-  const bodyOptions = {
-    file: blob,
-    sdk_source: 'onfido_web_sdk',
-    sdk_version: process.env.SDK_VERSION,
-    ...extraOptions
-  }
+const sendFile = (endpoint, data, token, onSuccess, onError) => {
+  data['sdk_source'] = 'onfido_web_sdk'
```

Comment:
modifying an object passed as parameter is prone to error if the expectation is not set right. the previous approach was better in this regard.